### PR TITLE
TR-1432 test level timeout outcome fix

### PR DESCRIPTION
**Pull request by augustas**
Fix outcome variable calculation on test level timeout. 
 
Related to : https://oat-sa.atlassian.net/browse/TR-1432
 
On test level timeout iterate through rest of test items and recalculate test outcome variables.
  
#### How to test
 
1. Create a test with test level timeout value set in authoring (can pick the test package from related jira ticket)
2. Create a delivery with that test
3. Launch the test
3. Do not answer all items and wait for timer to run out
4. Go to the back-office -> Results -> Choose your delivery -> Check the result. Outcome variables that sum MAXSCORE should have correct value. 

#### Dependencies
 
Requires :
 - [x] https://github.com/oat-sa/qti-sdk/pull/286

### CODE REVIEW

File: models/classes/runner/QtiRunnerService.php [hunk 1]
```diff
@@ -1503,7 +1503,7 @@ protected function onTimeout(RunnerServiceContext $context, AssessmentTestSessio
         switch ($timeOutException->getCode()) {
             case AssessmentTestSessionException::ASSESSMENT_TEST_DURATION_OVERFLOW:
                 \common_Logger::i('TIMEOUT: closing the assessment test session');
-                $session->endTestSession();
+                $session->moveThroughAndEndTestSession();
```
Comment (wazelin) [inline models/classes/runner/QtiRunnerService.php:1506]: Considering, this is a method, appearing in a very specific version of QTI-SDK, I strongly recommend explicitly requiring that version in `composer.json` as `^0.25` once it's been released.

**Review by wazelin — CHANGES_REQUESTED** [review]

File: models/classes/runner/QtiRunnerService.php (hunk 1 above)
Comment (augustas) [reply models/classes/runner/QtiRunnerService.php:1506]: Yep, good suggestion, I will do that when the tag will be released in `qti-sdk`

**Review by wazelin — COMMENTED** [review]
Will approve once the QTI-SDK library is released and required as an explicit dependency.

File: composer.json [hunk 2]
```diff
@@ -61,7 +61,7 @@
     "ext-dom": "*",
     "oat-sa/lib-test-cat": "2.3.7",
     "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
-    "qtism/qtism": "~0",
+    "qtism/qtism": "^0.24.11",
```
Comment (wazelin) [inline composer.json:64]: Please note this actually potentially prevents another issue of accidentally updating to a new major version, as the previous constraint didn't have an upper bound in the major version (legacy releases use 0.<major>.<minor/patch>` versioning).

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by emgolubev — APPROVED** [review]

**Comment by emgolubev** [comment]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: composer.json (hunk 2 above)
Comment (julien-sebire) [reply composer.json:64]: In fact, this is done in the extension-tao-itemqti. If you do this, you've to do it on both extensions...

**Review by julien-sebire — CHANGES_REQUESTED** [review]

**Review by julien-sebire — COMMENTED** [review]
Ok as long as you don't fix the version in here.

**Pull request merged**